fixed using block

heejaechang opened the pull request:
this fixes https://github.com/dotnet/roslyn/issues/22561

using block didn't properly created variable decls in the using block.

this require this PR to be in (https://github.com/dotnet/roslyn/pull/22435) since it addressed issues with newly re-enabled tests.

heejaechang commented:
@dotnet/analyzer-ioperation @AlekseyTs @dotnet/roslyn-compiler can you take a look?

AlekseyTs commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IVariableDeclaration.vb:
```diff
@@ -624,33 +624,58 @@ BC30053: Arrays cannot be declared with 'New'.
+  Value: 
+    null
+  Body: 
+    IBlockStatement (1 statements) (OperationKind.BlockStatement) (Syntax: 'Using c1 As ... End Using')
```
>IBlockStatement [](start = 4, length = 15)

Should this block be marked as IsImplicit? It looks like its parent is using the same syntax node.

heejaechang replied:
this is tracked by this issue - https://github.com/dotnet/roslyn/issues/22229

TyOverby commented on src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_IVariableDeclaration.vb:
```diff
@@ -1,4 +1,4 @@
-' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
```
@333fred: looks like the BOM issues are back.  Please investigate

333fred replied:
@heejaechang this was fixed. Have you updated?

heejaechang replied:
@TyOverby @333fred I didn't use the tool here. I manually updated the file and file was on wrong encoding and VS fixed it by moving to UTF-8 (it was saved in western english something encoding)

heejaechang commented:
fixed merge conflict and updated tests failing by this PR (there was another test that got enabled which had variable decl 0)

comment by @AlekseyTs about implicit block operation is tracked by this issue - https://github.com/dotnet/roslyn/issues/22229

heejaechang commented:
@dotnet/roslyn-infrastructure there seems a test harness issue on jenkins (ubuntu_14_debug_prtest) - https://ci.dot.net/job/dotnet_roslyn/job/master/job/ubuntu_14_debug_prtest/6888/

heejaechang commented:
retest ubuntu_14_debug_prtest please

heejaechang commented:
@jaredpar @jcouv @cston can I get one more code review and approval for ask mode?

jasonmalinowski commented:
@heejaechang Yep, @333fred is trying to figure that one out.

cston commented on src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb:
```diff
@@ -231,9 +231,9 @@ Namespace Microsoft.CodeAnalysis.Semantics
         End Function
 
-        Private Function GetVariableDeclarationStatementVariables(statement As BoundDimStatement) As ImmutableArray(Of IVariableDeclaration)
+        Private Function GetVariableDeclarationStatementVariables(declerations As ImmutableArray(Of BoundLocalDeclarationBase)) As ImmutableArray(Of IVariableDeclaration)
```
Typo: `declarations`

333fred commented:
@heejaechang can you add a test for getting just the `UsingStatementSyntax`, instead of the entire `UsingBlockSyntax`? I think this fix should take care of part of https://github.com/dotnet/roslyn/issues/22279#issuecomment-332956401, specifically the comment I linked. I'm expecting to see that just the UsingStatementSyntax would return something other than an empty IVariableDeclarationStatement.

heejaechang commented:
approval for checking in? @jaredpar @jcouv ?

heejaechang commented:
@333fred added the test you suggested

heejaechang commented:
ping?

jcouv commented:
I'll look now.

heejaechang commented:
@jcouv so am I approved to check in?

jcouv commented:
@jaredpar for ask-mode approval. Thanks

AlekseyTs commented:
@jinujoseph Can ask-mode approve IOperation related changes.
